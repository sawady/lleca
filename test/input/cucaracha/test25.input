
fun f(x : Bool) {
    if x {
        putChar(64)
    }
}

fun main() {
    if True {
        putChar(65)
    }
    if False {
        putChar(66)
    }

    b := True
    if b {
        putChar(67)
    }
    b := False
    if b {
        putChar(68)
    }
    putChar(69)
    f(True)
    putChar(70)
    f(False)
    putChar(71)
    f(b)
    putChar(72)
    b := True
    f(b)

    putChar(10)
}
