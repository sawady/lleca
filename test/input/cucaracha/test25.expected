Cons(
  Function(
    f,
    Cons(
      Parameter(
        x,
        Bool
      ),
      Nil
    ),
    Unit,
    Cons(
      StmtIf(
        ExprVar(
          x
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprConstNum(
                64
              ),
              Nil
            )
          ),
          Nil
        ),
        Nothing
      ),
      Nil
    )
  ),
  Cons(
    Function(
      main,
      Nil,
      Unit,
      Cons(
        StmtIf(
          ExprConstBool(
            True
          ),
          Cons(
            StmtCall(
              putChar,
              Cons(
                ExprConstNum(
                  65
                ),
                Nil
              )
            ),
            Nil
          ),
          Nothing
        ),
        Cons(
          StmtIf(
            ExprConstBool(
              False
            ),
            Cons(
              StmtCall(
                putChar,
                Cons(
                  ExprConstNum(
                    66
                  ),
                  Nil
                )
              ),
              Nil
            ),
            Nothing
          ),
          Cons(
            StmtAssign(
              b,
              ExprConstBool(
                True
              )
            ),
            Cons(
              StmtIf(
                ExprVar(
                  b
                ),
                Cons(
                  StmtCall(
                    putChar,
                    Cons(
                      ExprConstNum(
                        67
                      ),
                      Nil
                    )
                  ),
                  Nil
                ),
                Nothing
              ),
              Cons(
                StmtAssign(
                  b,
                  ExprConstBool(
                    False
                  )
                ),
                Cons(
                  StmtIf(
                    ExprVar(
                      b
                    ),
                    Cons(
                      StmtCall(
                        putChar,
                        Cons(
                          ExprConstNum(
                            68
                          ),
                          Nil
                        )
                      ),
                      Nil
                    ),
                    Nothing
                  ),
                  Cons(
                    StmtCall(
                      putChar,
                      Cons(
                        ExprConstNum(
                          69
                        ),
                        Nil
                      )
                    ),
                    Cons(
                      StmtCall(
                        f,
                        Cons(
                          ExprConstBool(
                            True
                          ),
                          Nil
                        )
                      ),
                      Cons(
                        StmtCall(
                          putChar,
                          Cons(
                            ExprConstNum(
                              70
                            ),
                            Nil
                          )
                        ),
                        Cons(
                          StmtCall(
                            f,
                            Cons(
                              ExprConstBool(
                                False
                              ),
                              Nil
                            )
                          ),
                          Cons(
                            StmtCall(
                              putChar,
                              Cons(
                                ExprConstNum(
                                  71
                                ),
                                Nil
                              )
                            ),
                            Cons(
                              StmtCall(
                                f,
                                Cons(
                                  ExprVar(
                                    b
                                  ),
                                  Nil
                                )
                              ),
                              Cons(
                                StmtCall(
                                  putChar,
                                  Cons(
                                    ExprConstNum(
                                      72
                                    ),
                                    Nil
                                  )
                                ),
                                Cons(
                                  StmtAssign(
                                    b,
                                    ExprConstBool(
                                      True
                                    )
                                  ),
                                  Cons(
                                    StmtCall(
                                      f,
                                      Cons(
                                        ExprVar(
                                          b
                                        ),
                                        Nil
                                      )
                                    ),
                                    Cons(
                                      StmtCall(
                                        putChar,
                                        Cons(
                                          ExprConstNum(
                                            10
                                          ),
                                          Nil
                                        )
                                      ),
                                      Nil
                                    )
                                  )
                                )
                              )
                            )
                          )
                        )
                      )
                    )
                  )
                )
              )
            )
          )
        )
      )
    ),
    Nil
  )
)
